DP_2_Problem_1.java

```diff
@@ -0,0 +1,73 @@
+//Brute force
+//Time Complexity : O(2^n * 3))
+//Space Complexity : Just the stack space O(1)
+// Have not bought leetcode premium yet
+// Problems :  No problem
+
+//Optimized DP
+//Time Complexity : O(n)
+//Space Complexity: O(m*n)
```

Comment:
Because colors are constant i.e. 3, there is only one factor affecting space, which is number of houses. So, space would also be O(N)